client/my-sites/pages/page/index.js

```diff
@@ -421,42 +423,154 @@ class Page extends Component {
 						siteUrl={ this.props.multisite && this.getSiteDomain() }
 					/>
 				</div>
-				{ ellipsisGridicon }
-				{ popoverMenu }
-				<ReactCSSTransitionGroup
-					transitionName="updated-trans"
-					transitionEnterTimeout={ 300 }
-					transitionLeaveTimeout={ 300 }
-				>
-					{ this.props.buildUpdateTemplate() }
-				</ReactCSSTransitionGroup>
+				{ ellipsisMenu }
+				{ shadowNotice }
 			</CompactCard>
 		);
 	}
 
+	changeShadowStatus( shadowStatus ) {
+		return this.props.onShadowStatusChange( this.props.page.global_ID, shadowStatus );
+	}
+
+	async performUpdate( { action, progressNotice, successNotice, errorNotice, undo } ) {
+		await this.changeShadowStatus( progressNotice );
+		try {
+			await action();
+			if ( undo === 'undo' ) {
+				// This update was actually undo. Reset the shadow status immediately
+				await this.changeShadowStatus( false );
+				return;
+			}
+			await this.changeShadowStatus( { ...successNotice, undo } );
+			if ( undo ) {
+				// If undo is offered, keep the success notice displayed indefinitely
+				return;
+			}
+		} catch ( error ) {
+			await this.changeShadowStatus( errorNotice );
+		}
+		// remove the success/error notice after 5 seconds
+		await sleep( 5000 );
+		await this.changeShadowStatus( false );
+	}
+
+	updatePostStatus( status ) {
+		const { page, translate } = this.props;
+
+		switch ( status ) {
+			case 'delete':
+				this.performUpdate( {
+					action: () => this.props.deletePost( page.site_ID, page.ID ),
+					progressNotice: {
+						status: 'is-error',
+						icon: 'trash',
+						text: translate( 'Deleting…' ),
+					},
+					successNotice: {
+						status: 'is-success',
+						text: translate( 'Page deleted.' ),
+					},
+					errorNotice: {
+						status: 'is-error',
+						text: translate( 'Failed to delete page.' ),
+					},
+				} );
+				return;
+
+			case 'trash':
+				this.performUpdate( {
+					action: () => this.props.trashPost( page.site_ID, page.ID, page ),
+					undo: page.status !== 'trash' ? 'restore' : 'undo',
+					progressNotice: {
+						status: 'is-error',
+						icon: 'trash',
+						text: translate( 'Trashing…' ),
+					},
+					successNotice: {
+						status: 'is-success',
+						text: translate( 'Page trashed.' ),
+					},
+					errorNotice: {
+						status: 'is-error',
+						text: translate( 'Failed to trash page.' ),
+					},
+				} );
+				return;
+
+			case 'restore':
+				this.performUpdate( {
+					action: () => this.props.restorePost( page.site_ID, page.ID ),
+					undo: page.status === 'trash' ? 'trash' : 'undo',
+					progressNotice: {
+						status: 'is-warning',
+						icon: 'history',
+						text: translate( 'Restoring…' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **29** times:
`translate( 'Restoring' )` *ES Score: 23*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/remove/update-post-status)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).